main/webservices/api/test_api.php

```diff
@@ -2,116 +2,531 @@
+        $courseList = [];
+        foreach($COURSE_CODES as $code) {
+            $course = CourseManager::create_course(['code' => $code, 'title' => $code, 'wanted_code' => $code ], UserManager::get_user_id_from_username(self::WEBSERVICE_USERNAME));
+            $courseList[] = $course['real_id'];
+        }
+        SessionManager::add_courses_to_session($modelSessionId, $courseList);
+
+        // call the webservice to create the new session from the model session, specifying a different value in the first extra field
+        $newSessionId = $this->integer($ACTION, [
+                'modelSessionId' => $modelSessionId,
+                'sessionName' => 'Name of the new session'.time(),
+                'startDate' => '2019-09-01 00:00',
+                'endDate' => '2019-12-31 00:00',
+                'extraFields' => [
+                    $EXTRA_FIELD_NAME => $EXTRA_FIELD_VALUE_FOR_NEW_SESSION
                 ],
-                'language' => 'english',
-                //'phone' => '',
-                //'expiration_date' => '',
-            ],
-        ]
-    );
+            ]
+        );
+
+        // assert both sessions are in the promotion
+        $promotionSessions = SessionManager::get_all_sessions_by_promotion($promotionId);
+        $this->assertSame(2, count($promotionSessions));
+        $this->assertArrayHasKey($newSessionId, $promotionSessions);
+        $this->assertArrayHasKey($modelSessionId, $promotionSessions);
+
+        // assert the new session has its own new value in the first extra field
+        $extraFieldValueModel = new ExtraFieldValue('session');
+        $extraFieldValue = $extraFieldValueModel->get_values_by_handler_and_field_variable($newSessionId, $EXTRA_FIELD_NAME);
+        $this->assertNotFalse($extraFieldValue);
+        $this->assertSame($EXTRA_FIELD_VALUE_FOR_NEW_SESSION, $extraFieldValue['value']);
+
+        // assert the model session still has its own original value in the first extra field
+        $extraFieldValue = $extraFieldValueModel->get_values_by_handler_and_field_variable($modelSessionId, $EXTRA_FIELD_NAME);
+        $this->assertNotFalse($extraFieldValue);
+        $this->assertSame($EXTRA_FIELD_VALUE_FOR_MODEL_SESSION, $extraFieldValue['value']);
+
+        // assert the new session has inherited the model session value in the second extra field
+        $extraFieldValue = $extraFieldValueModel->get_values_by_handler_and_field_variable($newSessionId, $SECOND_EXTRA_FIELD_NAME);
+        $this->assertNotFalse($extraFieldValue);
+        $this->assertSame($SECOND_EXTRA_FIELD_VALUE, $extraFieldValue['value']);
 
-    if ($response->getStatusCode() !== 200) {
-        throw new Exception('Entry denied with code : ' . $response->getStatusCode());
+        // assert the model session still has the same value in the second extra field
+        $extraFieldValue = $extraFieldValueModel->get_values_by_handler_and_field_variable($modelSessionId, $SECOND_EXTRA_FIELD_NAME);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Line exceeds maximum limit of 120 characters; contains 134 characters](https://app.codacy.com/app/chamilo/chamilo-lms/pullRequest?prid=4554450)